### DISCUSSION TITLE
FeatureHub Multi-Group Matching Permissions Issue

### DISCUSSION THREAD

**Discussion by jkim049** [featurehub-io/featurehub#1073] [General]
Hey peeps, I recently ran into an issue where folks in multiple groups were unable to perform actions on applications where the groups had matching permissions. Is this intended behavior or a bug?

I.E. A person is in group A which has all permissions for application Z & Y and in group B which has permissions for application Y & X. However, they were unable to modify flag values, add targeting rules, etc within application Y.

Note: We are on v1.6.2

**Comment by IrinaSouth** [comment]
Hi @jkim049, you should be able to change values for application Y in both groups, as long as permissions are set in the Group Permissions  page. Would you be able to provide some more details? I am unable to reproduce. Perhaps a screenshot from the Group Permissions page. 
Thanks

**Comment by jkim049** [comment]
Hi Irina, so the issue we saw was for non-admin users who had been in 2 groups that had overlapping permissions on certain applications. The first reported error was:
`ApiException 400:  Stack trace: ApiException 400: 
    at Object.d (https://featurehub.com/main-<string-redacted>js<line #>)`

Then when they attempted to modify flags they would receive an `An unexpected error occurred!` UI pop-up.

Unfortunately I was unable to reproduce this myself as a I have an account with admin privileges but this was prevalent across a few non-admin user accounts.

**Comment by IrinaSouth** [comment]
Hi @jkim049, thanks for providing extra details. I think I managed to reproduce. To confirm the scenario, it looks like this happens if there is Group1 with permission `Read` and Group2 with permissions `Read` and `Change Values` and a user appears to be in both groups. Then on the attempt to change value, it throws 400 error. We are going to include a fix for the next release (in this case of overlapping permissions, we will allow a user to edit values since this permission in one of the groups).
Thanks

**Comment by IrinaSouth** [comment]
+ opened issue to track https://github.com/featurehub-io/featurehub/issues/1075
